src/main/java/com/fasterxml/jackson/dataformat/xml/deser/QNameDeserializer.java

```diff
@@ -0,0 +1,33 @@
+package com.fasterxml.jackson.dataformat.xml.deser;
+
+import javax.xml.namespace.QName;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+
+public class QNameDeserializer extends JsonDeserializer {
+    JsonDeserializer<?> originalDeserializer;
+    public QNameDeserializer(JsonDeserializer<?> deserializer) {
+        originalDeserializer = deserializer;
+    }
+
+    @Override
+    public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JacksonException {
+        QName qName = (QName) originalDeserializer.deserialize(p, ctxt);
+
+        if (qName.getLocalPart().indexOf(":") > 0) {
+            String prefix = qName.getLocalPart().split(":")[0];
+            String localPart = qName.getLocalPart().split(":")[1];
+            String namespace = ((FromXmlParser)ctxt.getParser()).getStaxReader().getNamespaceContext().getNamespaceURI(prefix);
```

Comment:
This won't work if content is buffered, so I am not sure this approach is valid unfortunately.

Reply:
Humm.. interesting.. do you have a simple test case for the Buffered use case? 

I am using the same design in our solution so, even if it is not a viable solution here, I would love to improve the code in our solution. Or at least know em when exactly it will fail.

Reply:
It's a side-effect of processing, not a feature to enable: if you create a POJO with `@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)`, property values will likely be buffered. Similarly for polymorphic types if Type Id is not the first value deserialized.
Basically any case where stream-order is not the same as the order in which values are needed will be buffered.

Reply:
So, the correct way to solve this would be to change the token streams to include namespaces and prefixes available at each one of them?

Reply:
Yes. `TokenBuffer`, however, is not XML specific and cannot (should not) be changed.
2.13 allows replacement of buffer implementation so we could have `XmlTokenBuffer` sub-class (or such) -- XML backend was the reason to add this -- but there's then the question of how to access information. Probably `FromXmlParser` and `XmlTokenBuffer` could implement an interface (to be added) for additional accessors.

I have not, unfortunately, had time to take this approach any further but this would be a very valuable improvement and unblock work on solving multiple issues.
So I would be happy to help you or anybody who had time to look into improvements in this area.